sdk/storage/src/blob/blob/responses/get_blob_metadata_response.rs

```diff
@@ -0,0 +1,50 @@
+use azure_core::errors::AzureError;
+use azure_core::headers::{
+    date_from_headers, etag_from_headers, request_id_from_headers, server_from_headers,
+};
+use azure_core::{Metadata, RequestId};
+use bytes::Bytes;
+use chrono::{DateTime, Utc};
+use http::HeaderMap;
+use std::convert::TryFrom;
+
+#[derive(Debug, Clone)]
+pub struct GetBlobMetadataResponse {
+    pub request_id: RequestId,
+    pub etag: String,
+    pub server: String,
+    pub date: DateTime<Utc>,
+    pub metadata: Metadata,
+}
+
+impl TryFrom<&HeaderMap> for GetBlobMetadataResponse {
+    type Error = AzureError;
+
+    fn try_from(headers: &HeaderMap) -> Result<Self, Self::Error> {
+        debug!("headers == {:#?}", headers);
+
+        let mut metadata = Metadata::new();
+        headers
+            .iter()
+            .filter(|header| header.0.as_str().starts_with("x-ms-meta-"))
+            .for_each(|header| {
```

Comment:
nit: can you deconstruct the `header` binding as `(key, value)`?

Reply:
Good idea!

Reply:
Even better, it's probably best to move this code to the `Metadata` struct.